File: 2-4/2-4-TS/dogs.ts
/*
 * This is a program that uses mrCoxallStack class for dog items.
 * @author  Jakub Malhotra
 * @version 3.0 
 * @since   2024-10-21
 */

import MrCoxallStack from './mrCoxallStack.ts'

// dog Stack
const dogStack = new MrCoxallStack()
console.log(`Dog items: ${dogStack.showStack}`)
console.log(`The size of the stack is: ${dogStack.size}`)
console.log(`Is the stack empty? ${dogStack.isEmpty}`)
// push items
dogStack.pushItem('poodle')
dogStack.pushItem('retriever')
dogStack.pushItem('beagle')
// print full stack
console.log(`Dog items: ${dogStack.showStack}`)
// pop and show an item
const topDog = dogStack.popItem();
console.log(`Popped dog item: ${topDog}`);
// display stack after popping item
console.log(`Dog items: ${dogStack.showStack}`)
console.log(`The size of the stack is: ${dogStack.size}`)
console.log(`Is the stack empty? ${dogStack.isEmpty}`)
// peek and display an item
console.log(`The top dog is: ${dogStack.peekItem}`)
console.log(`The size of the stack is: ${dogStack.size}`)
// empty the stack and show it
console.log(`Emptying the stack ...`)
dogStack.clearStack
console.log(`The size of the stack is: ${dogStack.size}`)
console.log(`Is the stack empty? ${dogStack.isEmpty}`)

console.log('\nDone.')

File: 2-4/2-4-Java/Dogs.java
/*
 * This is a program that uses MrCoxallStack class for dog items.
 * @author  Jakub Malhotra
 * @version 3.0
 * @since   2024-10-21
 */

public final class Dogs {

    // Private constructor to prevent instantiation of utility class
    private Dogs() {
        throw new IllegalStateException("This is the main function");
    }

    /**
     * The main method uses MrCoxallStack for dog items.
     *
     * @param args command line arguments
     */
    public static void main(final String[] args) {
        // Dog Stack
        MrCoxallStack dogStack = new MrCoxallStack();
        System.out.println("Dog items: " + dogStack.getShowStack());
        System.out.println("The size of the stack is: " + dogStack.getSize());
        System.out.println("Is the stack empty? " + dogStack.isEmpty());

        // Push items
        dogStack.pushItem("poodle");
        dogStack.pushItem("retriever");
        dogStack.pushItem("beagle");

        // Print full stack
        System.out.println("Dog items: " + dogStack.getShowStack());

        // Pop and show an item
        String topDog = dogStack.popItem();
        System.out.println("Popped dog item: " + topDog);

        // Display stack after popping item
        System.out.println("Dog items: " + dogStack.getShowStack());
        System.out.println("The size of the stack is: " + dogStack.getSize());
        System.out.println("Is the stack empty? " + dogStack.isEmpty());

        // Peek and display an item
        System.out.println("The top dog is: " + dogStack.peekItem());
        System.out.println("The size of the stack is: " + dogStack.getSize());

        // Empty the stack and show it
        System.out.println("Emptying the stack ...");
        dogStack.clearStack();
        System.out.println("The size of the stack is: " + dogStack.getSize());
        System.out.println("Is the stack empty? " + dogStack.isEmpty());

        System.out.println("\nDone.");
    }
}
